src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -785,6 +786,7 @@ FileSegmentPtr FileSegment::getSnapshot(const FileSegmentPtr & file_segment, std
     snapshot->downloaded_size = file_segment->getDownloadedSizeUnlocked(segment_lock);
     snapshot->download_state = file_segment->download_state;
     snapshot->segment_kind = file_segment->getKind();
+    snapshot->is_unbound = file_segment->is_unbound;
```

Comment:
Let's also add it here https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/System/StorageSystemFilesystemCache.cpp#L64